index.js

```diff
@@ -1,13 +1,29 @@
-const express = require('express');
-const path = require('path');
-const app = express();
-const port = process.env.PORT || 8080;
-
-app.use(express.static(path.join(__dirname, 'client/build')));
-app.get('*', (require, response) => {
-  response.sendFile(path.join(__dirname, 'client/build/index.html'));
-});
+const jsonServer = require('json-server');
+const server = jsonServer.create();
+const router = jsonServer.router('db.json');
+const middlewares = jsonServer.defaults();
 
-app.listen(port, () => {
-  console.log('Server is running');
+server.use(
+  jsonServer.rewriter({
+    '/api/*': '/$1',
+  })
+);
+server.use(middlewares);
+server.use(router);
+server.listen(4000, () => {
```

Comment:
Your proxy (client/package.json) targets port 8080. You changed the port to 4000 which is the reason the proxy doesn't work.